### DISCUSSION TITLE
Bundling Platform types with plugins

### DISCUSSION THREAD

**Discussion by hgmich** [Vector35/binaryninja-api#2975] [Q&A] [answered]
Per some discussion in the Binary Ninja Slack, it seems the only way to have types/variables/system calls associated with a Platform is using platform type definitions, which currently reside in either `$APPDIR/types/platform` or `$USERDIR/types/platform`. I'm working on a platform plugin - is there any way to bundle these into my plugin, or are they not picked up elsewhere?

I've tried adding the types to a type library as an alternative, but even when explicitly associating those types with a platform, they don't get picked up as part of the platform's system calls, types and variables. This is particularly a problem for system calls, which seemingly aren't possible to define in other ways currently.

It sounds like there were plans to add APIs to handle this, but this issue relating to that hasn't been updated in a while: #1936

**Comment by rssor** [comment, marked as answer]
This actually looks like I just forgot to close out issues when I added support. When defining the Platform it's now possible to provide a path to a header file to use that you keep inside your plugin directory:

https://github.com/Vector35/binaryninja-api/blob/6812c973c9fa9b4ad642ab81856c05f87bd6fcc4/python/platform.py#L60

**Reply by hgmich** [reply]
Exactly what I was looking for, thanks!
